site/content/integrate/apps/interactivity/dynamic-form-example/_index.md

```diff
@@ -0,0 +1,544 @@
+
+### Manifest
+
+When our App is installed, we will return this Manifest to define our App:
+
+```ts
+app.get('/manifest', (req, res) => {
+    const manifest = {
+        app_id: 'ticket-form-example',
+        display_name: "Create tickets using forms",
+        homepage_url: 'https://github.com/mattermost/mattermost-plugin-apps',
+        app_type: 'http',
+        icon: 'icon.png',
+        root_url: `http://localhost:3000`,
+        requested_permissions: [
+            'act_as_bot', // This is necessary to create public posts as our bot
+        ],
+        requested_locations: [
+            '/post_menu', // This is necessary to render elements in the post menu
+        ],
+    } as AppManifest;
+
+    res.json(manifest);
+});
+```
+
+With our server running, we can then install our App in Mattermost using this command:
+
+`/apps install http http://localhost:3000/manifest`
+
+### Bindings
+
+The App will receive a request to provide UI elements to show to the user, whenever the user visits a different channel. This allows us to dynamically show different UI elements or slash commands in each channel. We need to handle this request being sent to the `/bindings` endpoint.
+
+```ts
+app.post('/bindings', (req, res) => {
+    const call: AppCallRequest = req.body;
+
+    // Use the user id and channel id to dynamically return certain UI elements. Though for this example, we won't use these value.
+    const channelID = call.context.channel_id;
+    const userID = call.context.acting_user_id;
+
+    const bindings: AppBinding[] = [{
+        location: '/post_menu',
+        bindings: [
+            {
+                location: 'create-ticket',
+                icon: 'create-ticket.png',
+                label: 'Create ticket',
+                call: {
+                    path: '/forms/create-ticket-post-menu',
+                    expand: {
+                        post: 'summary',
+                    },
+                },
+            },
+        ],
+    }];
+
+    const callResponse: AppCallResponse = {
+        type: 'ok',
+        data: bindings,
+    };
+
+    res.json(callResponse);
+});
+
+// When static files are fetched from our App, the request paths are prefixed with `/static`, so our files `icon.png` and `create-ticket.png` will be requested as `static/icon.png` and `static/create-ticket.png`. If we have all of our static assets in one directory, we can serve all of them like this:
+const staticDirectory = './static';
+app.use('/static', express.static(staticDirectory));
+```
+
+We now have an item in the post dropdown menu that says "Create ticket", along with the icon located at `static/create-ticket.png` displayed next to the label. When the post menu item is clicked, our App will receive a [Call]({{< ref "call" >}}) at the endpoint `/forms/create-ticket-from-post/submit`. If we return a form in our response, a modal will be shown to the user with our form in the modal.
```

Comment:
1/5 - to be comprehensive, it usually make sense to also mention the negative case ("and if we do not return a form" - you will get an error since the path points to "/forms" or smth. else)